pkg/cloud/services/loadbalancer/loadbalancer.go

```diff
@@ -42,20 +43,37 @@ import (
 const (
 	networkPrefix   string = "k8s-clusterapi"
 	kubeapiLBSuffix string = "kubeapi"
+	resolvedMsg     string = "ControlPlaneEndpoint.Host is not an IP address, using the first resolved IP address"
 )
 
 const loadBalancerProvisioningStatusActive = "ACTIVE"
 
+// We wrap the net.LookupHost function in a variable to allow overriding it in unit tests.
+//
+//nolint:gocritic
+var lookupHost = func(host string) ([]string, error) {
+	return net.LookupHost(host)
```

Comment:
A few things here:

This should probably use Resolver.LookupHost() as documented in net.LookupHost(), but we can't because we don't currently pass the context into this ReconcileLoadBalancer. We should most likely also do that.

I seem to recall we have a forcing function coming soon which will require us to use Context everywhere. So we don't miss this one, which isn't obvious without reading the godocs, could we update it to use Resolver.LookupHost anyway and create a context explicitly to pass in?

Secondly, does this have the correct behaviour when given an IP address? Either way, I wonder if we should check if it's an IP address before calling the resolver and just return it.

Reply:
1. Using `net.DefaultResolver.LookupHost`: I agree, consider it done in the next patchset.


2. Does this have the correct behaviour when given an IP address? yes:

```golang
// LookupHost looks up the given host using the local resolver.
// It returns a slice of that host's addresses.
func (r *Resolver) LookupHost(ctx context.Context, host string) (addrs []string, err error) {
	// Make sure that no matter what we do later, host=="" is rejected.
	if host == "" {
		return nil, &DNSError{Err: errNoSuchHost.Error(), Name: host, IsNotFound: true}
	}
	if _, err := netip.ParseAddr(host); err == nil {             <----------- HERE
		return []string{host}, nil
	}
	return r.lookupHost(ctx, host)
}
```